internal/git/releases.go

```diff
@@ -61,14 +61,25 @@ func (g *Git) CreateRelease(releaseInfo releases.ReleasesInfo) error {
 				return fmt.Errorf("failed to run goreleaser: %w", err)
 			}
 		} else {
+			tagName := github.String(tag)
 			_, _, err = g.client.Repositories.CreateRelease(context.Background(), os.Getenv("GITHUB_REPOSITORY_OWNER"), getRepo(), &github.RepositoryRelease{
-				TagName:         github.String(tag),
+				TagName:         tagName,
 				TargetCommitish: github.String(commitHash),
 				Name:            github.String(fmt.Sprintf("%s - %s - %s", lang, tag, environment.GetInvokeTime().Format("2006-01-02 15:04:05"))),
 				Body:            github.String(fmt.Sprintf(`# Generated by Speakeasy CLI%s`, releaseInfo)),
 			})
 			if err != nil {
-				return fmt.Errorf("failed to create release: %w", err)
+				if release, _, err := g.client.Repositories.GetReleaseByTag(context.Background(), os.Getenv("GITHUB_REPOSITORY_OWNER"), getRepo(), *tagName); err == nil && release != nil {
+					fmt.Println(fmt.Sprintf("a github release with tag %s already existing ... skipping publishing", *tagName))
+					fmt.Println(fmt.Sprintf("to publish this version again delete the github tag and release"))
+					if _, ok := outputs[fmt.Sprintf("publish_%s", lang)]; ok {
+						outputs[fmt.Sprintf("publish_%s", lang)] = "false"
```

Comment:
I realize we do this in other places but I don't think its ideal to take in a loosely typed map and then infer things about its contents, like that it has keys in the format `publish_<lang>`. There are many ways to improve that--one option would be to add a helper like `func shouldPublishLanguage(lang string, outputs map[string]string)`

Reply:
Wouldn't the function just do the same thing. Check if it exists if it does set it to false? In this case